### PR TITLE
Added string xor

### PR DESCRIPTION
**Pull request by christopherlagerhult**

### CODE REVIEW

File: source/system/String.ooc [hunk 1]
```diff
@@ -250,6 +250,13 @@ String: class {
 		other free()
 		result
 	}
+	operator ^ (other: This) -> This {
+		Debug error(other size == 0, "Size of input must be greater than zero")
+		result := ""
+		for (i in 0 .. this _buffer size)
+			result += this[i] ^ other[i % other size]
```
Comment (marcusnaslund) [inline source/system/String.ooc]: `&=`, otherwise this will leak

File: test/system/StringTest.ooc [hunk 2]
```diff
@@ -213,6 +213,17 @@ StringTest: class extends Fixture {
...
+		this add("xor", func {
+			key := "!98đsv"
+			input := "void main() {
+					result := true
+				}"
+
+			alteredInput := input ^ key
+			expect(alteredInput, is notEqual to(input))
+			restoredInput := alteredInput ^ key
+			expect(restoredInput, is equal to(input))
```
Comment (marcusnaslund) [inline test/system/StringTest.ooc]: `(alteredInput, restoredInput) free()`

Comment (christopherlagerhult) [inline test/system/StringTest.ooc]: Oh were back in ooc :)

Comment (marcusnaslund) [inline test/system/StringTest.ooc]: oh the joy

**Comment by marcusnaslund** [comment]
What will you be using this for? Should we make sure it's ported to c++?

**Pull request closed without merging**